### PR TITLE
move Drawer and GithubCorner to their own components

### PR DESCRIPTION
**Pull request by sylvainpolletvillard**
This will help for the future work on redesigning the whole drawer. I changed a bit the toggle, let me know what you think of it.

I also think we should split the CSS by component so that things can be manageable at scale. This requires to change the current PostCSS setup to process styles processed by Svelte, is it possible ?

### CODE REVIEW

File: app/components/App/App.svelte.html [hunk 1]
```diff
@@ -1,43 +1,18 @@
...
-        <path d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3,112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z" fill="currentColor" class="octo-body"></path>
-      </svg>
-    </a>
-    <style>.github-corner:hover .octo-arm{animation:octocat-wave 560ms ease-in-out}@keyframes octocat-wave{0%,100%{transform:rotate(0)}20%,60%{transform:rotate(-25deg)}40%,80%{transform:rotate(10deg)}}@media (max-width:500px){.github-corner:hover .octo-arm{animation:none}.github-corner .octo-arm{animation:octocat-wave 560ms ease-in-out}}</style>
-    <!-- End GH corner component -->
+    <GithubCorner/>
```
Comment (RyanZim) [inline app/components/App/App.svelte.html]: Should be `<GithubCorner />` (space before slash).

File: app/components/Drawer/Drawer.svelte.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+				<svg xmlns="http://www.w3.org/2000/svg" width="10" height="10" viewBox="0 0 10 10">
+					<path d="m 8.746788,5.4163099 0,4.1630977 -8.74250548,0 0,-7.9098857 4.99571738,0 0,0.8326195 -4.16309782,0 0,6.2446471 7.07726632,0 0,-3.3304786 0.8326196,0 z m 1.2489295,-4.99571746 -4.574412,0 1.67981,1.66523926 -2.9045934,2.94331 1.177324,1.1773241 L 8.2784396,3.2631557 9.9957175,5 l 0,-4.57940756 z"/>
+				</svg>
+			</a>
+		</label>
+		{{/each}}
+	</div>
+</div>
+
+<script src="Drawer.js"></script>
```
Comment (RyanZim) [inline app/components/Drawer/Drawer.svelte.html]: Needs newline at EOF

File: app/components/GithubCorner/GithubCorner.svelte.html [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	@media (max-width: 500px) {
+		.github-corner:hover .octo-arm {
+			animation: none
+		}
+
+		.github-corner .octo-arm {
+			animation: octocat-wave 560ms ease-in-out
+		}
+	}
+</style>
```
Comment (RyanZim) [inline app/components/GithubCorner/GithubCorner.svelte.html]: Needs Newline at EOF

File: package.json [hunk 4]
```diff
@@ -23,7 +23,7 @@
     "eslint": "eslint . --ignore-path .prettierignore",
     "format": "npm run eslint -- --fix && npm run prettier -- --write && sort-package-json",
     "test": "npm run eslint && npm run prettier -- --list-different && npm run build-production",
-    "prettier": "prettier \"**/*.{js,md,css}\"",
+    "prettier": "prettier \"**/*.{js,md,css}\" --write",
```
Comment (RyanZim) [inline package.json]: Why this change? This breaks the scripts. `format` script is for writing changes, `test` script is for the CI, and should fail if the files aren't correct (shouldn't write).

**Comment by RyanZim** [comment]
> I also think we should split the CSS by component so that things can be manageable at scale. This requires to change the current PostCSS setup to process styles processed by Svelte, is it possible ?

File: package.json (hunk 4 above)
Comment (sylvainpolletvillard) [reply package.json]: sorry, I did not mean to push this

**Comment by RyanZim** [comment]
@sylvainpolletvillard Pulled this in and pushed a couple of spacing tweaks. If you're OK with that, I can merge.

**Comment by RyanZim** [comment]
Just realized there were tabs here; fixed that and added `.editorconfig` file.

**Comment by sylvainpolletvillard** [comment]
It's weird that Prettier did not fix the indentation. OK to merge

**Comment by RyanZim** [comment]
@sylvainpolletvillard Prettier currently doesn't format the svelte components. PR queued up after this one to change that.

**Pull request merged**